core/src/tasks/profile/sweep.ts

```diff
@@ -0,0 +1,53 @@
+import { api } from "actionhero";
+import { QueryTypes } from "sequelize";
+import { CLSTask } from "../../classes/tasks/clsTask";
+import { Profile } from "../../models/Profile";
+import { plugin } from "../../modules/plugin";
+
+export class ProfileSweep extends CLSTask {
+  constructor() {
+    super();
+    this.name = "profiles:sweep";
+    this.description =
+      "Remove profiles that no longer have any directly mapped properties";
+    this.frequency = 1000 * 60 * 30; // every half hour
+    this.queue = "profiles";
+    this.inputs = {};
+  }
+
+  async runWithinTransaction() {
+    const limit = parseInt(
+      (await plugin.readSetting("core", "runs-profile-batch-size")).value
+    );
+
+    // Get profiles that don't have directlyMapped properties and whose exports have settled
+    const profiles: { id: string }[] = await api.sequelize.query(
+      `
+      SELECT "id" FROM "profiles" WHERE "state"='ready' 
+        AND 0 = (
+          SELECT count("exports"."id") FROM "exports" 
+            WHERE "exports"."profileId"="profiles"."id" 
+            AND "exports"."state" IN ('pending', 'processing')
+        ) AND "id" NOT IN (
+          SELECT DISTINCT("profileId") FROM "profileProperties" 
+          JOIN properties ON "properties"."id"="profileProperties"."propertyId" 
+          WHERE "properties"."directlyMapped"=true AND "rawValue" IS NOT NULL
+        ) LIMIT ${limit};
+      `,
```

Comment:
This is complex query, but it's tested well!
Also, confirm that it's OK on SQLite here - https://app.circleci.com/pipelines/github/grouparoo/grouparoo/10107/workflows/48841cdc-003b-426e-9a64-d03f39f01fe0/jobs/215309